src/trace-plugin-interface.js

```diff
@@ -0,0 +1,246 @@
+'use strict';
+var TraceLabels = require('./trace-labels.js');
+var cls = require('./cls.js');
+var constants = require('./constants.js');
+var is = require('is');
+
+/**
+ * This file describes an interface for third-party PluginAPIs to enable tracing
+ * for arbitrary modules.
+ */
+
+/**
+ * An object that is associated with a single child span. It exposes
+ * functions for adding labels to or closing the associated span.
+ */
+function ChildSpan(agent, context) {
+  this.agent = agent;
+  this.context = context;
```

Comment:
Could we rename this to `privateSpan` or maybe `span`. Context is a pretty overloaded term. (Span is too but maybe it's better?)